CLAUDE.md

```diff
@@ -0,0 +1,137 @@
+
+#### `builder/`
+- **Purpose**: Registry and factory system for dynamic object creation
+- **Key Features**: Type registration, building patterns, configuration-driven construction
+- **Documentation**: See `zetta_utils/builder/CLAUDE.md`
+
+### Specialized Modules
+
+#### `geometry/`
+- **Purpose**: 3D geometry utilities for bounding boxes, vectors, spatial operations
+- **Key Features**: Bbox operations, striding, mask centering, vector math
+
+#### `convnet/`
+- **Purpose**: Convolutional neural network utilities and inference runners
+- **Key Features**: Model architectures, inference pipelines, utilities
+
+#### `training/`
+- **Purpose**: PyTorch training utilities and data loading
+- **Key Features**: Custom data loaders, samplers, training workflows
+
+#### `augmentations/`
+- **Purpose**: Data augmentation for training and inference
+- **Key Features**: Tensor augmentations, misalignment simulation, imgaug integration
+
+#### `parsing/`
+- **Purpose**: Configuration parsing and state management
+- **Key Features**: CUE parsing, JSON handling, neuroglancer state management
+
+### Utility Modules
+
+#### `common/`
+- **Purpose**: Shared utilities and helper functions
+- **Key Features**: Context managers, path utilities, timers, signal handling
+
+#### `db_annotations/`
+- **Purpose**: Database-backed annotation management
+- **Key Features**: Collections, layers, annotation storage and retrieval
+
+#### `api/`
+- **Purpose**: External API interfaces
+- **Key Features**: Version management, API endpoints
+
+#### `cli/`
+- **Purpose**: Command-line interface tools
+- **Key Features**: Task management CLI, main entry points
+
+#### `ng/`
+- **Purpose**: Neuroglancer integration utilities
+- **Key Features**: Link builders, state management
+
+#### `run/`
+- **Purpose**: Runtime utilities and resource management
+- **Key Features**: Garbage collection, resource monitoring
+
+#### `viz/`
+- **Purpose**: Visualization utilities and widgets
+- **Key Features**: Rendering, interactive widgets
+
+### Data Processing Modules
+
+#### `tensor_mapping/`
+- **Purpose**: Tensor mapping and transformation utilities
+- **Key Features**: Coordinate mapping, tensor transformations
+
+#### `message_queues/`
+- **Purpose**: Message queue abstractions and serialization
+- **Key Features**: Queue interfaces, message serialization
+
+#### `distributions/`
+- **Purpose**: Distribution utilities for distributed computing
+- **Key Features**: Common distribution patterns
+
+#### `mazepa_addons/`
+- **Purpose**: Extensions and add-ons for mazepa framework
+- **Key Features**: Additional utilities for mazepa workflows
+
+#### `mazepa_layer_processing/`
+- **Purpose**: Layer processing operations for mazepa workflows
+- **Key Features**: Annotation postprocessing, operation protocols
+
+#### `cloud_management/`
+- **Purpose**: Cloud resource management utilities
+- **Key Features**: Cloud provider integrations
+
+## General Guidelines
```

Comment:
Maybe add a Code Style Requirements section, or point to the formatter settings. Some examples:
  * Use Black formatter with a 99 character line length limit
  * Follow PEP 8 naming conventions
  * Use type hints for all function signatures
  * Maximum number of locals per function: 25
  * Add docstrings for all public functions using Sphinx-RTD style

Reply:
Performance vs. maintainability trade-offs:
* benchmark current vs proposed implementation
* default to maintainability unless critical code path

Reply:
Breaking-changes handling:
* prefer backward-compatibility
* implement deprecation warnings first